diff --git a/setup.py b/setup.py
index c2b6446..8a29348 100644
--- a/setup.py
+++ b/setup.py
@@ -97,6 +97,7 @@ def build_library():
         finally:
             os.chdir(currentDir)
     elif host_OS == "Windows":
+        print("IN BUILD setup.py spice")
         currentDir = os.getcwd()
         try:
             destination = os.path.join(cspice_dir, "src", "cspice")
@@ -106,11 +107,14 @@ def build_library():
             shutil.copy(makeBat, destination)
             # run the script
             os.chdir(destination)
+            print("about to run make dynamic")
             windows_build = subprocess.Popen("makeDynamicSpice.bat", shell=True)
+            print("finished make dynamic")
             status = windows_build.wait()
             if status != 0:
                 raise BaseException('{0}'.format(status))
         except BaseException as error:
+            print("RAN INTO EXCEPTION: ", error)
             sys.exit("Build failed with: {0}".format(error.args))
             pass
         finally:
@@ -125,6 +129,24 @@ def move_to_root_directory():
             if host_OS == "Linux" or host_OS == "Darwin":
                 os.rename(os.path.join(cspice_dir, 'lib', 'spice.so'), destination)
             elif host_OS == "Windows":
+                print("")
+                print("")
+                print("####################################################")
+                print("in move_to_root for windows")
+                print("cspice_dir: " + cspice_dir)
+                print("")
+                print("what is in cspice_dir")
+                print([x for x in os.listdir(cspice_dir)])
+                print("")
+                print("what is in cspice_dir/src")
+                print([x for x in os.listdir(os.path.join(cspice_dir, 'src'))])
+                print("")
+                print("what is in cspice_dir/src/cspice")
+                print([x for x in os.listdir(os.path.join(cspice_dir, 'src', 'cspice'))])
+                print("")
+                print("####################################################")
+                print("")
+                print("")
                 os.rename(os.path.join(cspice_dir, 'src', 'cspice', 'cspice.dll'), destination)
         except BaseException as e:
             sys.exit('{0} file not found, what happend?: {1}'.format(sharedLib, e))
@@ -160,7 +182,9 @@ def windows_method():
             return
         else:
             # Build the DLL
+            print('IN WINDOWS METHOD')
             build_library()
+            print("OUTSIDE BUIOLD_LIBRARY")
             # Move to correct location (root of the distribution)
             move_to_root_directory()
 
